udtf.go

```diff
@@ -0,0 +1,493 @@
+package duckdb
+
+/*
+#include <stdlib.h>
+#include <duckdb.h>
+
+void udf_bind_row(duckdb_bind_info info);
+void udf_bind_chunk(duckdb_bind_info info);
+
+void udf_init(duckdb_init_info info);
+void udf_local_init(duckdb_init_info info);
+
+void udf_row_callback(duckdb_function_info, duckdb_data_chunk);  // https://golang.org/issue/19837
+void udf_chunk_callback(duckdb_function_info, duckdb_data_chunk);  // https://golang.org/issue/19837
+
+void udf_destroy_data(void *);
+
+typedef void (*init)(duckdb_function_info);  // https://golang.org/issue/19835
+typedef void (*bind)(duckdb_function_info);  // https://golang.org/issue/19835
+typedef void (*callback)(duckdb_function_info, duckdb_data_chunk);  // https://golang.org/issue/19835
+*/
+import "C"
+
+import (
+	"database/sql"
+	"runtime"
+	"runtime/cgo"
+	"unsafe"
+)
+
+type (
```

Comment:
Let's move `ColumnMetaData`, `CardinalityData`, and `ThreadedTableSourceInitData` to `udf_utils.go`.

Reply:
Yes, I can see giving this plan a shot.
I am just a bit conflicted about the name `table_source.go`, since a source emits data, and appending data to a table sounds like the contrary. 🤔 
What about something like `chunk_manager`? We never really talk about the table, only data chunks.

Reply:
Hmm, I think table source would work since you are essentially taking the union of 2 tables, the appending data is just the specific implementation of how that is done.

> since a source emits data, and appending data to a table sounds like the contrary

True, the way I thought about is, is that you have this source (user), and a sink (duckdb), and go-duckdb is just the connector. But I see your point. I will think about it a bit more.